packages/protocol/contracts/common/Proxy.sol

```diff
@@ -88,6 +95,7 @@ contract Proxy {
     payable
     onlyOwner
   {
+    require(_isContract(implementation), "Invalid contract address");
```

Comment:
Actually, let's move this check to `_setImplementation` to catch this both at initialization time, and any other time the implementation changes.